### PR TITLE
Lighting efficiency updates

### LINKED ISSUE #553: Lighting causes serious lag when qdel() fails en masse.
#### Description of issue
Lighting overlays (/obj/effect/light) fail to qdel() and cause massive lag when they suddenly hard delete all at once.

#### Difference between expected and actual behavior
Lighting overlays should qdel cleanly.

#### Steps to reproduce 
Join a round and cause a massive lighting update via power cut/admin secrets panel.

#### Length of time in which bug has been known to occur
Ever since the new lighting system was added.

#### Server revision & Game ID
Server revision: master - 2016-12-15
46199492c467c887da6a84abfba88b910e1aa388 
Game ID: bL4-dFQY

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of the issue being looked at sooner.

- [X] Issue could be reproduced at least once
- [X] Issue could be reproduced by different players
- [X] Issue could be reproduced in multiple rounds
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] [Couldn't find an existing issue about this](https://github.com/Baystation12/Baystation12/issues)

### PR DESCRIPTION
**Pull request by Loganbacca**
Runtime prevention in Destroy()
Faster light off proc

Hopefully resolves #553

### CODE REVIEW

File: code/modules/europa/planar_lighting/light_effect.dm [hunk 1]
```diff
@@ -54,7 +57,7 @@
 		T.lumcount = -1
 		T.affecting_lights -= src
 	affecting_turfs.Cut()
-	. = .. ()
+	. = ..()
```
Comment (Loganbacca) [inline code/modules/europa/planar_lighting/light_effect.dm]: Does this actually matter?

Comment (Techhead0) [inline code/modules/europa/planar_lighting/light_effect.dm]: I tested `.. ()` and it works. Apparently.

File: code/modules/europa/planar_lighting/light_effect.dm [hunk 2]
```diff
@@ -36,9 +36,12 @@
...
 /obj/effect/light/Destroy()
-	moved_event.unregister(holder, src)
-	dir_set_event.unregister(holder, src)
-	destroyed_event.unregister(holder, src)
+	if(moved_event)
+		moved_event.unregister(holder, src)
+	if(dir_set_event)
+		dir_set_event.unregister(holder, src)
+	if(destroyed_event)
+		destroyed_event.unregister(holder, src)
```
Comment (PsiOmegaDelta) [inline code/modules/europa/planar_lighting/light_effect.dm]: Hm. Even registrations should be removed automatically on qdel.
Wonder if I've missed a case of some sort.

Comment (Loganbacca) [inline code/modules/europa/planar_lighting/light_effect.dm]: Dunno... I couldn't recreate the issues locally, so I haven't really been able to test if the problem in #553 is gone.

**Pull request merged**